cdap-ui/app/cdap/components/AbstractWidget/FunctionDropdownAliasWidget/FuctionDropdownRow.tsx

```diff
@@ -74,59 +84,106 @@ class FunctionDropdownRow extends AbstractRow<IFunctionDropdownRowProps, IKeyVal
     field: '',
     func: '',
     alias: '',
+    hasCondition: false,
+    condition: '',
   };
 
   /**
    * Sample input: alias:Avg(fieldName)
    */
   public componentDidMount() {
-    const [alias, fn] = this.props.value.split(':');
-    const { func, field } = this.extractFunctionAndAlias(fn);
+    const [alias, fn, conditionValue] = this.props.value.split(':');
+    const { func, field, hasCondition, condition } = this.extractFunctionAliasAndCondition(
+      fn,
+      conditionValue
+    );
 
     this.setState({
       field,
       func,
       alias,
+      hasCondition,
+      condition,
     });
   }
 
-  private extractFunctionAndAlias(fn) {
+  private extractFunctionAliasAndCondition(fn, conditionValue) {
     const defaultResponse = {
       func: '',
       field: '',
+      hasCondition: false,
+      condition: '',
     };
 
     if (!fn) {
       return defaultResponse;
     }
 
-    const openBracketIndex = fn.indexOf('(');
-    const closeBracketIndex = fn.indexOf(')');
+    const fnOpenBracketIndex = fn.indexOf('(');
+    const fnCloseBracketIndex = fn.indexOf(')');
+    const condiOpenBracketIndex = conditionValue.indexOf('(');
+    const condiCloseBracketIndex = conditionValue.indexOf(')');
 
-    if (openBracketIndex === -1 || closeBracketIndex === -1) {
+    if (fnOpenBracketIndex === -1 || fnCloseBracketIndex === -1) {
       return defaultResponse;
     }
 
+    if (conditionValue) {
+      return {
+        func: fn.substring(0, fnOpenBracketIndex),
+        field: fn.substring(fnOpenBracketIndex + 1, fnCloseBracketIndex),
+        hasCondition: true,
+        condition: conditionValue.substring(condiOpenBracketIndex + 1, condiCloseBracketIndex + 1),
+      };
+    }
+
     return {
-      func: fn.substring(0, openBracketIndex),
-      field: fn.substring(openBracketIndex + 1, closeBracketIndex),
+      func: fn.substring(0, fnOpenBracketIndex),
+      field: fn.substring(fnOpenBracketIndex + 1, fnCloseBracketIndex),
+      hasCondition: false,
+      condition: '',
     };
   }
 
+  private handleCondition = (type: StateKeys, e) => {
+    if (type === 'func') {
+      const targetValueObj = this.props.dropdownOptions.filter(
+        (el) => typeof el === 'object' && el.value === e.target.value
```

Comment:
Is this the condition to check if it is an conditional aggregate function? When user chooses `MaxIf` or `AvgIf` functions how do we determine to show condition textbox?

Reply:
The reason why the check is in line 155 is because IDropdownOption can be a string, number or IComplexDropdown (object). If the IDropdownOption comes as a string or a number, hasCondition state should be false. On the other way if the IDropdownOption comes as an Object, with hasOwnProperty() we check if 'hasCondition' exists in the Object because this is an optional property.

If we extract the check inside the filter method we can not handle when IDropdownOption comes as a string or number.